### PR TITLE
MB-2964 Update system architecture for Prime API

### PR DESCRIPTION
**Pull request by sandy-wright**
## Description

This PR updates the architecture for this codebase to be more extensible as we add more tests for the Prime & Support APIs. It also adds a few utilities to make it easier to switch environments, handle mTLS certs, and other quality of life updates.

## Reviewer Notes

Some old code was moved to fit in with the new organization, but it isn't currently functional. My bandaid for this was to stick it in the file called `fixme.py`, but I would appreciate feedback on whether or not it should just be removed for now (it'll still be hanging around in version control, after all).

I also set up a small "proof of concept" load test for the Office app that implements some of the code in the `fixme.py` and shows it working with the new architecture. We could consider adding a ticket to update the rest of the older load testing code in a similar way.

## Setup

First, remove the previous virtual environment (if you have previously checked out the project):

```sh
make teardown
```

Create a new python virtual environment and install all of the dependencies:

```sh
make setup
```

Activate the virtual environment:

```sh
. .venv/bin/activate
```

Now, navigate to your `mymove` directory and activate the local server:

```sh
make db_dev_e2e_populate && make server_run
```

Back in your `milmove_load_testing` directory, run the default Prime load testing command (there will be a number of exceptions here because the `SupportTasks` class is still abstract, but there will be successful requests too):

```sh
make load_test_prime
```

Or try running only the `PrimeUser` class for an exception-free experience:

```sh
locust -f locustfiles/prime.py --host local PrimeUser
```

Open http://localhost:8089 to execute the load test from the web interface. Or, alternatively, run it through the CLI:

```sh
locust -f locustfiles/prime.py --host local --headless --users 10 --hatch-rate 2 PrimeUser
```

Sit back, watch the logs, look at the charts, and enjoy! 

Once you are done testing, click the big red "Stop" button in the web interface (if using), use a keyboard interrupt to cancel the process on the command line, and `deactivate` your virtual environment.

## Code Review Verification Steps

* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [MB-2964](https://dp3.atlassian.net/browse/MB-2964) for updating the system architecture for this repo.
* [MB-2716](https://dp3.atlassian.net/browse/MB-2716) for adding the `fetch_mto_updates` task.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## Description
```
Comment (abbyoung) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: ❤️

Comment (abbyoung) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Thank you for adding the PR template!

File: tasks/prime.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ class PrimeTasks(CertTaskSet):
     tags where appropriate to make filtering for custom tests easier.
     """
 
-    @tag("mto")
+    @tag("mto", "fetchMTOUpdates")
```
Comment (shimonacarvalho) [inline tasks/prime.py:28]: Are locust tags case-sensitive?

File: locustfiles/prime.py [hunk 3]
```diff
@@ -13,12 +13,13 @@ class PrimeUser(MilMoveHostMixin, HttpUser):
...
-    domain = MilMoveDomain.PRIME
-    is_api = True
+    domain = MilMoveDomain.PRIME  # the base domain for the host
+    is_api = True  # if True, uses the api base domain in deployed environments
+    # cert_kwargs are used by CertTaskSet for verifying requests:
     cert_kwargs = PRIME_CERT_KWARGS  # TODO will need to be handled differently for staging/experimental
 
-    wait_time = between(1, 9)
-    tasks = {PrimeTasks: 1}
+    wait_time = between(0.25, 9)  # the time period to wait in between tasks (in seconds, accepts decimals and 0)
```
Comment (shimonacarvalho) [inline locustfiles/prime.py:21]: Ah decimals!

File: tasks/prime.py (hunk 2 above)
Comment (sandy-wright) [reply tasks/prime.py:28]: I wasn't sure, but I just tested it and they are! We may end up deciding we want a different tagging scheme at some point, I think

**Pull request merged**